src/ui/QuerySuggestPreview/QuerySuggestPreview.ts

```diff
@@ -178,7 +178,13 @@ export class QuerySuggestPreview extends Component implements IComponentBindings
   }
 
   private buildResultPreview(suggestionText: string, element: HTMLElement, rank: number): ISearchResultPreview {
-    $$(element).addClass('coveo-preview-selectable');
+    element.classList.add('coveo-preview-selectable');
+    const resultLink = element.querySelector(Component.computeSelectorForType(ResultLink.ID)) as HTMLElement;
+    if (resultLink) {
+      element.setAttribute('aria-label', resultLink.textContent);
```

Comment:
Shouldn't that be directly inside the ResultLink component logic ?

Reply:
I'm gonna check it out, but I think `aria-label` might not be allowed when `role` is set to `header`. If it's allowed, I'll add that into `ResultLink`.